### ISSUE TITLE
kubectl create service missing a selector flag?

### ISSUE THREAD

**Issue by je-al** [kubernetes/kubernetes#39424]
Hi, this is a feature request, unless I'm missing something.

I'm running:
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"2016-12-14T00:57:05Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.1", GitCommit:"82450d03cb057bab0950214ef122b67c83fb11df", GitTreeState:"clean", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}
```

I was following this [quick-start guide](http://kubernetes.io/docs/user-guide/quick-start/#launching-a-simple-application-and-exposing-it-to-the-internet), but since I'm running in minikube I wanted to specify a service using NodePort instead of a LoadBalancer. Since the expose command does not provide a flag (AFAICT) to specify the actual nodeport to use (the one the host exposes) (which is being discussed in  #25478), I tried to use `kubectl create service nodeport` only to find out (again AFAICT) it has **no way to specify the selector**.

You _can_ patch it afterwards (i.e. `kubectl patch svc --type=json -p='[{"op":"replace","path":"/spec/selector","value":{"run":"my-nginx"}}]' my-nginx`) but, unless this is the intended UX, it seems like the selector flag should be available.

**Comment by kargakis** [comment]
@kubernetes/sig-cli-misc

**Comment by pwittrock** [comment]
@ymqytw

**Comment by ymqytw** [comment]
@je-al It seems that `kubectl create service nodeport` doesn't support that. So this is a feature request.

An workaround for you is that:
Create an service using `kubectl create -f`. You can specify the selector in the config file. Please take a look at [service doc](http://kubernetes.io/docs/user-guide/services/). There are several examples.
```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: 2017-01-06T18:10:49Z
  labels:
    app: mynp
    env: dev
  name: mynp
spec:
  clusterIP: 10.0.235.132
  ports:
  - nodePort: 31975
    port: 80
    protocol: TCP
    targetPort: 9376
  selector:
    app: mynp
    env: dev
  type: NodePort
```

**Comment by kargakis** [comment]
I don't think we want to add any more flags for these commands cc @deads2k

You can pipe to `kubectl set selector`

**Comment by je-al** [comment]
@kargakis is the selector sub-command WIP or a proposal?
I don't have it available on kubectl, nor can I see it in the docs.

**Comment by kargakis** [comment]
> @kargakis is the selector sub-command WIP or a proposal? I don't have it available on kubectl, nor can I see it in the docs.

It's in master, will be included in 1.6.

**Comment by je-al** [comment]
Oh, ok then, if there actually is a "prescribed" way to do it (for my part) we can close this.
